ibtk/include/ibtk/HierarchyMathOps.h

```diff
@@ -535,16 +535,16 @@ class HierarchyMathOps : public SAMRAI::tbox::DescribedClass
      * \see resetLevels
      */
     void grad(int dst_idx,
-              SAMRAI::tbox::Pointer<SAMRAI::pdat::FaceVariable<NDIM, double> > dst_var,
+              const SAMRAI::pdat::FaceVariable<NDIM, double>* dst_var,
               bool dst_cf_bdry_synch,
               double alpha,
               int src1_idx,
-              SAMRAI::tbox::Pointer<SAMRAI::pdat::CellVariable<NDIM, double> > src1_var,
-              SAMRAI::tbox::Pointer<HierarchyGhostCellInterpolation> src1_ghost_fill,
+              const SAMRAI::pdat::CellVariable<NDIM, double>* src1_var,
+              HierarchyGhostCellInterpolation* src1_ghost_fill,
               double src1_ghost_fill_time,
               double beta = 0.0,
               int src2_idx = invalid_index,
-              SAMRAI::tbox::Pointer<SAMRAI::pdat::FaceVariable<NDIM, double> > src2_var = NULL,
+              const SAMRAI::pdat::FaceVariable<NDIM, double>* src2_var = NULL,
```

Comment:
in contrast, since these are optional arguments, we should use pointers here. Can you use `nullptr`? How did `NULL` sneak back in?

Reply:
Changed to `nullptr`. I don't think that this PR introduced `NULL` here ...

Reply:
I'm not sure how NULL got back in - we need to avoid that (especially when we mix pointers and integers)

Reply:
Agh I must have dropped the commit where I fixed that.

Reply:
Ah, I did fix it -- I think you are looking at an outdated commit?